test/models/asyncapi_test.js

```diff
@@ -978,6 +984,88 @@ describe('AsyncAPIDocument', function() {
   });
   /* eslint-enable sonarjs/cognitive-complexity */
 
+  describe('#stringify()', function() {
+    it('should stringify simple document', async function() {
+      const doc = await parser.parse(simpleInputJSON, { path: path.join(__filename, '../../') });
+      const stringified = AsyncAPIDocument.stringify(doc);
+      expect(stringified).to.be.equal(simpleOutputJSON);
+    });
+
+    it('should stringify document with circular references', async function() {
+      const doc = await parser.parse(circularYAML, { path: path.join(__filename, '../../') });
+      const stringified = AsyncAPIDocument.stringify(doc);
+      expect(stringified).to.be.equal(circularOutputYAML);
+    });
+  });
+
+  describe('#parse()', function() {
+    it('should parse stringified simple document', async function() {
+      const parsedDoc = await parser.parse(simpleInputJSON, { path: path.join(__filename, '../../') });
+      const doc = AsyncAPIDocument.parse(simpleOutputJSON);
+      expect(JSON.stringify(doc.json())).to.be.equal(JSON.stringify(parsedDoc.json()));
+    });
+
+    it('should parse stringified document with circular references', async function() {
+      // Test circular references to ensure that every circular reference has this same reference after parsing
+      const result = AsyncAPIDocument.parse(circularOutputYAML);
+
+      // not testing on a model level as required xParserCircle value is added before model construction so we need to test through calling parser function
+      expect(result.hasCircular()).to.equal(true);
+
+      // we want false here, even though this schema has some circular refs in some props, it is not circular, but just specific items
+      expect(result.components().schema('RecursiveSelf').isCircular()).to.equal(false);
```

Comment:
I find way of reusing variables easier to read in terms of the expected tests:
```suggestion
      const resultComponents = result.components();
      const recursiveSelfComponent = resultComponents.schema('RecursiveSelf');
      expect(recursiveSelfComponent.isCircular()).to.equal(false);
```

If it is just one or two recurring times it is used I don't mind, but when it's 2+ times I find this easier to read and to maintain. What do you think?

This needs to be applied below as well of course 🙂

Reply:
On the one hand ok, but on the other hand not, because if I make a variable for each object, I will have to remember that the variable points to the element, and now I have a full track of what compares to what. But I will stay with the current form..